src/java/org/apache/cassandra/db/commitlog/CompressedSegment.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.cassandra.db.commitlog;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.channels.FileChannel;
+
+import org.apache.cassandra.config.DatabaseDescriptor;
+import org.apache.cassandra.io.FSWriteError;
+import org.apache.cassandra.io.compress.ICompressor.WrappedArray;
+import org.apache.cassandra.io.util.FileUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/*
+ * A single commit log file on disk. Manages creation of the file and writing mutations to disk,
+ * as well as tracking the last mutation position of any "dirty" CFs covered by the segment file. Segment
+ * files are initially allocated to a fixed size and can grow to accomidate a larger value if necessary.
+ */
+public class CompressedSegment extends CommitLogSegment
+{
+    private static final Logger logger = LoggerFactory.getLogger(CompressedSegment.class);
+
+    private final FileChannel channel;
+    WrappedArray compressedArray = new WrappedArray(new byte[1024]);
+    ByteBuffer compressedBuffer = ByteBuffer.wrap(compressedArray.buffer);
+    
+    static final int COMPRESSED_MARKER_SIZE = SYNC_MARKER_SIZE + 4;
+
+    /**
+     * Constructs a new segment file.
+     *
+     * @param filePath  if not null, recycles the existing file by renaming it and truncating it to CommitLog.SEGMENT_SIZE.
+     */
+    CompressedSegment(String filePath)
+    {
+        super(filePath);
+        try
+        {
+            channel = logFileAccessor.getChannel();
+            channel.write((ByteBuffer) buffer.duplicate().flip());
```

Comment:
Since this is a FileChannel, the doc does specify that "a write operation will return only after writing all of the r requested bytes" (http://docs.oracle.com/javase/7/docs/api/java/nio/channels/WritableByteChannel.html).

Fewer bytes may only be written when the method throws an exception, which will not be helped by checking.